mmeds/util.py

```diff
@@ -721,32 +722,25 @@ def mmeds_to_MIxS(file_fp, out_file, skip_rows=0, unit_column=None):
                 f.write('\t'.join([header] + list(map(str, df[col1][col2].tolist()))) + '\n')
 
 
-def log(text, testing=False, write_file=fig.MMEDS_LOG):
-    """ Write provided text to the log file. """
+def log(text, testing=False, log_type='Debug'):
+    """ Pass provided text to the MmedsLogger. If given a dictionary, format it into a string. """
     if isinstance(text, dict):
         log_text = '\n'.join(["{}: {}".format(key, value) for (key, value) in text.items()])
     elif isinstance(text, list):
         log_text = '\n'.join(list(map(str, text)))
     else:
         log_text = str(text)
 
-    if testing:
-        with open('/tmp/mmeds_log.txt', 'a+') as f:
-            f.write('{}: {}\n'.format(datetime.now(), log_text))
-    else:
-        with open(write_file, 'a+') as f:
-            f.write('{}: {}\n'.format(datetime.now(), log_text))
+    if log_type == 'Debug':
+        logger = MMEDSLog('Debug_logger', testing)
+    elif log_type == 'SQL':
+        logger = MMEDSLog('SQL_logger', testing)
+    logger.debug(log_text)
```

Comment:
This log function by calling logger.debug will cause any logged statements to be logged at Debug level. I'm guessing you've done it this way to avoid having to change code in a lot of other files for calling the logger, but it might be helpful to think about what should be logged at Info vs Debug level. For instance, I usually use Info level logging for code that is at a higher level and less likely to be called frequently. Debug level logging I'll put at lower levels and include values of local variables.

Reply:
Your assessment of this is largely correct. I suppose now is the appropriate time to do an overview of what is getting logged where. Most of the logs relate to whatever I was recently working on, rather than working into a general plan